tests/validator/validator.cpp

```diff
@@ -1764,3 +1771,47 @@ TEST(Validator, unitComplexCycle)
 
     EXPECT_EQ_ERRORS(expectedErrors, v);
 }
+
+TEST(Validator, duplicatedCellMLUnitsOnCiElement)
+{
+    const std::string math =
+        "<math xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns=\"http://www.w3.org/1998/Math/MathML\">\n"
+        "  <ci cellml:units=\"dimensionless\" cellml:units=\"second\">B</ci>\n"
```

Comment:
I'd thought that we didn't define units on the `ci` elements, only the `cn` ones?

Reply:
It's not defined by the spec. I think.  The spec. doesn't say that you can't do that and somebody is free to write that sort of thing whether it is a validation error is at this point undefined.  I wouldn't mind someone pointing to the section in the spec. that refutes this though.

Reply:
In other sections of the validator/parser we do not permit random stuff to be present in the cellml ... not sure why this would be any different?  It's not (currently) explicitly forbidden, but if it is allowed, then we need to read it and validate it against the units defined for that variable elsewhere in the code.  I don't like the idea of having two possibly conflicting definitions permitted, and the user being unaware which is actually being used.